lib/ansible/modules/cloud/vmware/vmware_host_sriov.py

```diff
@@ -0,0 +1,331 @@
+  vmware_host_sriov:
+    hostname: "{{ vcenter_hostname }}"
+    username: "{{ vcenter_username }}"
+    password: "{{ vcenter_password }}"
+    esxi_hostname: "{{ esxi1 }}"
+    validate_certs: no
+    vmnic: vmnic0
+    sriov_on: false
+    num_virt_func: 0
+"""
+
+RETURN = r"""
+host_sriov_diff:
+    description:
+    - contains info about SR-IOV staus on vmnic before, after and requested changes
+    - sometimes vCenter slowly update info, as result "after" contains same info as "before"
+      need to run again in check_mode or reboot host, as ESXi requested
+    returned: always
+    type: dict
+    "sample": {
+        "changed": true,
+        "diff": {
+            "after": {
+                "host_test": {
+                    "sriovActive": false,
+                    "sriovEnabled": true,
+                    "maxVirtualFunctionSupported": 63,
+                    "numVirtualFunction": 0,
+                    "numVirtualFunctionRequested": 8,
+                    "rebootRequired": true,
+                    "sriovCapable": true
+                }
+            },
+            "before": {
+                "host_test": {
+                    "sriovActive": false,
+                    "sriovEnabled": false,
+                    "maxVirtualFunctionSupported": 63,
+                    "numVirtualFunction": 0,
+                    "numVirtualFunctionRequested": 0,
+                    "rebootRequired": false,
+                    "sriovCapable": true
+                }
+            },
+            "changes": {
+                "host_test": {
+                    "numVirtualFunction": 8,
+                    "rebootRequired": true,
+                    "sriovEnabled": true
+                }
+            }
+        }
+    }
+"""
+
+
+try:
+    from pyVmomi import vim
+except ImportError:
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware import vmware_argument_spec, PyVmomi
+from ansible.module_utils._text import to_native
+from time import sleep
+
+
+class VmwareAdapterConfigManager(PyVmomi):
+    """Class to configure SR-IOV settings"""
+
+    def __init__(self, module):
+        super(VmwareAdapterConfigManager, self).__init__(module)
+        cluster_name = self.params.get("cluster_name", None)
+        esxi_host_name = self.params.get("esxi_hostname", None)
+
+        self.vmnic = self.params.get("vmnic", None)
+        self.sriov_on = self.params.get("sriov_on", None)
+        self.num_virt_func = self.params.get("num_virt_func", None)
+
+        self.hosts = self.get_all_host_objs(
+            cluster_name=cluster_name, esxi_host_name=esxi_host_name
+        )
+        if not self.hosts:
+            self.module.fail_json(msg="Failed to find host system.")
+        self.results = {"before": {}, "after": {}, "changes": {}}
+
+    def set_host_state(self):
+        """Check ESXi host configuration"""
+        change_list = []
+        changed = False
+        for host in self.hosts:
```

Comment:
is this loop applicable only Cluster name is passed right?

Reply:
I got this error
```
{"failed": true, "msg": "Module alias error: internal error: required and default are mutually exclusive for num_virt_func"}
```

Reply:
I didnt get this. Did you make any changes to loop and got this error?

Reply:
sorry, ignore my previous post - wrong thread.
> is this loop applicable only Cluster name is passed right?
yes, you are right